tools/alive_parser.cpp

```diff
@@ -394,8 +417,13 @@ static unique_ptr<Instr> parse_binop(string_view name, token op_token) {
   auto &type = parse_type();
   auto &a = parse_operand(type);
   parse_comma();
-  auto &type2 = try_parse_type(type);
+
+  // 2nd binop argument handling
+  auto type2 = std::ref(try_parse_type(type));
+  if (op_token == EXTRACTVALUE)
+    type2 = std::ref(get_int_type(64));
```

Comment:
why override type2?

Reply:
If I don't override it and we don't want second argument of `extractvalue` to mention a type with the constant, it will automatically assume second type to be first argument's type (which in this case is aggregate type and incorrect). Therefore, I made second argument of `extractvalue` be implicitly parse as 64 bit integer.

Reply:
Then just use try_parse_type() without passing type. If the type is given it should be respected, instead of overriding it.